test/unit/helpers/cache.js

```diff
@@ -0,0 +1,100 @@
+'use strict';
+
+var chai = require('chai');
+var expect = require('chai').expect;
+var sinon = require('sinon');
+var redis = require('redis');
+
+var cache = require('../../../helpers/cache');
+
+describe('cache', function () {
+
+	it('is not enabled', function () {
+		cache.connect(false, {}, {}, function (err, result) {
+			expect(result.cacheEnabled).to.eq( false );
+			expect(result.client).to.be.null;
+		});
+	});
+
+	it('is enabled redis client callback ready', function () {
+
+		var logger = {
+			'info': function (info) {expect(info).to.eq('App connected with redis server');}
+		};
+
+		var mockredisclient = {
+			'on': function (type, cb) {
+				if ( 'ready' === type ) {
+					cb();
+				}
+			}
+		};
+
+		var mockedRedis = sinon.mock(redis);
+		mockedRedis.expects('createClient').once().withArgs(sinon.match.any).returns(mockredisclient);
+		cache.connect(true, {}, logger, function (err, result) {
+			expect(result.cacheEnabled).to.eq( true );
+			expect(result.client).to.be.not.null;
+
+			mockedRedis.restore();
+			mockedRedis.verify();
```

Comment:
Sinon mock's `verify` function internally calls `restore` function. So, here, `mockedRedis.restore()` is redundant.

Reply:
Thanks for pointing this out... I was not aware of that!

Updated the tests to create in the beforeEach() method and verify in the afterEach() method.